app/services/ContributionOphanService.scala

```diff
@@ -25,104 +25,98 @@ trait ContributionOphanService {
+  */
+class OphanServiceWithLogging(service: OphanService, infoMessage: LoggingContext => String)
+  extends ContributionOphanService with TagAwareLogger with RuntimeClassUtils {
 
-  import AcquisitionSubmissionBuilder.ops._
   import cats.instances.future._
 
   def submitAcquisition[A : AcquisitionSubmissionBuilder : ClassTag](a: A)(
     implicit ec: ExecutionContext,
     tags: LoggingTags,
     request: MetaDataRequest[_]
-  ): EitherT[Future, String, AcquisitionSubmission] =
-    EitherT.fromEither[Future](a.asAcquisitionSubmission).bimap(
+  ): EitherT[Future, OphanServiceError, AcquisitionSubmission] =
+    service.submit(a).bimap(
       err => {
-        error(err)
+        error(s"${err.getMessage} - contributions session id ${request.sessionId}")
         err
       },
       submission => {
-        info(s"Acquisition submission created from instance of ${runtimeClass[A]} - $submission")
+        val context = LoggingContext(runtimeClass[A], request, submission)
+        info(infoMessage(context))
         submission
       }
     )
 }
 
-class DefaultContributionOphanService(service: OphanService)
-  extends ContributionOphanService with TagAwareLogger with RuntimeClassUtils {
-
-  import cats.instances.future._
-  import actions.CommonActions._
-  import AcquisitionSubmissionBuilder.ops._
+object OphanServiceWithLogging {
 
-  private def sendSubmission(submission: AcquisitionSubmission)(implicit executionContext: ExecutionContext) = {
-    import submission._
-    service.submit(acquisition, ophanIds.browserId, ophanIds.viewId, ophanIds.visitId)
-      .bimap(err => s"Failed to submit acquisition event to Ophan - ${err.getMessage}", _ => submission)
-  }
-
-  def submitAcquisition[A : AcquisitionSubmissionBuilder : ClassTag](a: A)(
-    implicit ec: ExecutionContext,
-    tags: LoggingTags,
-    request: MetaDataRequest[_]
-  ): EitherT[Future, String, AcquisitionSubmission] =
-    EitherT.fromEither[Future](a.asAcquisitionSubmission).flatMap(sendSubmission)
-      .bimap(
-        err => {
-          error(s"$err - contributions session id ${request.sessionId}")
-          err
-        },
-        submission => {
-          info(
-            s"Acquisition submission created from an instance of ${runtimeClass[A]} and " +
-            s"successfully submitted to Ophan - contributions session id ${request.sessionId}"
-          )
-          submission
-        }
-      )
+  /**
+    * Context required to for useful logging messages on a successful acquisition submission.
+    */
+  private[services] case class LoggingContext(
+    runtimeClass: Class[_],
+    request: MetaDataRequest[_],
+    submission: AcquisitionSubmission
+  )
+
+  def io(uri: HttpUrl)(implicit client: OkHttpClient): OphanServiceWithLogging =
+    new OphanServiceWithLogging(OphanService(uri), { ctx =>
+      s"Acquisition submission created from an instance of ${ctx.runtimeClass} and " +
```

Comment:
Maybe this log entry should be generated within the `OphanServiceWithLogging` submission handler (since `OphanService` is guaranteed to at least create a submission), and the parameter provided here can be an optional additional info message to log whatever happens after the submission gets built?